example/examplechecker.py

```diff
@@ -1,128 +1,190 @@
+    port = 80
+
+    def login(self, credentials) -> None:
+        res = self.http_post("/login", json=credentials)
+        assert_equals(res.status_code, 200)
+
+    def register_and_login(self, credentials) -> None:
+        res = self.http_post("/register", json=credentials)
+        assert_equals(res.status_code, 200)
 
-    def putflag(self):  # type: () -> None
+        self.login(credentials)
+
+    def generate_credentials(self) -> Dict[str, str]:
+        credentials = {
+            "username": secrets.token_urlsafe(12),
+            "password": secrets.token_urlsafe(16),
+        }
+        self.team_db[self.flag] = credentials
+        return credentials
+
+    def putflag(self) -> None:
         """
-            This method stores a flag in the service.
-            In case multiple flags are provided, self.flag_idx gives the appropriate index.
-            The flag itself can be retrieved from self.flag.
-            On error, raise an Eno Exception.
-            :raises EnoException on error
-            :return this function can return a result if it wants
-                    if nothing is returned, the service status is considered okay.
-                    the preferred way to report errors in the service is by raising an appropriate enoexception
+        This method stores a flag in the service.
+        In case multiple flags are provided, self.flag_idx gives the appropriate index.
+        The flag itself can be retrieved from self.flag.
+        On error, raise an Eno Exception.
+        :raises EnoException on error
         """
         if self.flag_idx == 0:
-            self.team_db[sha256ify(self.flag)] = self.flag
+            credentials = self.generate_credentials()
+            self.register_and_login(credentials)
+
+            res = self.http_post("/notes", json={"note": self.flag,})
+            assert_equals(res.status_code, 200)
         elif self.flag_idx == 1:
-            self.global_db[
-                "{}_{}".format(self.address, self.flag)
-            ] = "Different place for different flag_idx"
+            credentials = self.generate_credentials()
+            self.register_and_login(credentials)
+
+            res = self.http_post("/profile/status", json={"status": self.flag})
+            assert_equals(res.status_code, 200)
         else:
             raise ValueError(
                 "Call_Idx {} exceeds the amount of flags. Not supported.".format(
                     self.flag_idx
                 )
             )
 
-    def getflag(self):  # type: () -> None
+    def getflag(self) -> None:
         """
         This method retrieves a flag from the service.
         Use self.flag to get the flag that needs to be recovered and self.round to get the round the flag was placed in.
         On error, raise an EnoException.
         :raises EnoException on error
-        :return this function can return a result if it wants
```

Comment:
This is still valid, isn't it? Sometimes returning here makes it easier to write. At least it should stay documented
(Or is it invalid now?)

Reply:
According to the type annotations that was never valid to begin with, I didn't even know there was code to handle this. But considering that we usually want the error reason (e.g. https://github.com/enowars/enowars4-scoreboard/issues/2) exceptions should always be preferred over return values imo

Reply:
The type annotations were for this specific checker, I guess.
Some of last year's services used returns, for example:
https://github.com/enowars/enowars3-service-voting/blob/7acde8546daea3b577af6364d4c4084829737664/checker/checker.py#L67
However, error-reason is a valid point!
Let's drop it from the docu and keep it undocumented feature for now?

Reply:
Aah, you are right. In the BaseChecker they are still correctly annotated with `-> Optional[Result]`.

It could make sense to include a warning when returning a result other than `OK` and suggest using an Exception with a reason for the error instead.

Reply:
Follow-up in #57